components/model_analyzer/tb_dcgm_types/gpu_device_factory.py

```diff
@@ -47,34 +48,58 @@ def init_all_devices(self, dcgmPath=None):
             Absolute path to dcgm shared library
         """
 
-        if numba.cuda.is_available():
-            logger.info("Initiliazing GPUDevice handles")
-            structs._dcgmInit(dcgmPath)
-            dcgm_agent.dcgmInit()
-
-            # Start DCGM in the embedded mode to use the shared library
-            dcgm_handle = dcgm_agent.dcgmStartEmbedded(
-                structs.DCGM_OPERATION_MODE_MANUAL)
-
-            # Create a GPU device for every supported DCGM device
-            dcgm_device_ids = dcgm_agent.dcgmGetAllSupportedDevices(dcgm_handle)
-            for device_id in dcgm_device_ids:
-                device_atrributes = dcgm_agent.dcgmGetDeviceAttributes(
-                    dcgm_handle, device_id).identifiers
-                pci_bus_id = device_atrributes.pciBusId.upper()
-                device_uuid = device_atrributes.uuid
-                device_name = device_atrributes.deviceName
+        if self._model_analyzer_backend == 'dcgm':
+            if numba.cuda.is_available():
+                logger.info("Initiliazing GPUDevice handles using DCGM")
+                structs._dcgmInit(dcgmPath)
+                dcgm_agent.dcgmInit()
+
+                # Start DCGM in the embedded mode to use the shared library
+                dcgm_handle = dcgm_agent.dcgmStartEmbedded(
+                    structs.DCGM_OPERATION_MODE_MANUAL)
+
+                # Create a GPU device for every supported DCGM device
+                dcgm_device_ids = dcgm_agent.dcgmGetAllSupportedDevices(dcgm_handle)
+                for device_id in dcgm_device_ids:
+                    device_atrributes = dcgm_agent.dcgmGetDeviceAttributes(
+                        dcgm_handle, device_id).identifiers
+                    pci_bus_id = device_atrributes.pciBusId.upper()
+                    device_uuid = device_atrributes.uuid
+                    device_name = device_atrributes.deviceName
+                    try:
+                        gpu_device = GPUDevice(device_name, device_id, pci_bus_id,
+                                            device_uuid)
+                    except TorchBenchAnalyzerExceptionGPUUnavailable as e:
+                        logger.info(e)
+                        continue
+                    self._devices.append(gpu_device)
+                    self._devices_by_bus_id[pci_bus_id] = gpu_device
+                    self._devices_by_uuid[device_uuid] = gpu_device
+
+                dcgm_agent.dcgmShutdown()
+        else:
+            import pynvml
+
+            logger.info("Initializing GPUDevice handles using NVML")
+            pynvml.nvmlInit()
+
+            # Create a GPU device for every supported NVML device
+            nvml_device_count = pynvml.nvmlDeviceGetCount()
+            for device_id in range(nvml_device_count):
+                handle = pynvml.nvmlDeviceGetHandleByIndex(device_id)
+                device_name = pynvml.nvmlDeviceGetName(handle)
```

Comment:
We need to decode the name here
```suggestion
                device_name = pynvml.nvmlDeviceGetName(handle).decode('ascii')
                pci_bus_id = pynvml.nvmlDeviceGetPciInfo(handle).busId.decode('ascii')
                device_uuid = pynvml.nvmlDeviceGetUUID(handle).decode('ascii')
```

Otherwise they won't pass the check at https://github.com/pytorch/benchmark/blob/main/components/model_analyzer/tb_dcgm_types/gpu_device.py#L51

Reply:
Sorry. I should do more compatibility checks. Testing on pynvml 11.4.1 now. Will fix them soon.